cert/cert_v2.go

```diff
@@ -231,7 +231,16 @@ func (c *certificateV2) MarshalForHandshakes() ([]byte, error) {
 		//TODO: panic on nil rawDetails
 		b.AddBytes(c.rawDetails)
 
-		// Skipping the curve and public key since those come across in a different part of the handshake
+		// Skipping public key since those come across in a different part of the handshake
+
+		//todo is curve skippable? I don't think so?
+
+		// Add the curve only if its not the default value
```

Comment:
We can transmit the byte but we will need to assert the curve we used in handshaking matches the curve present here.

The intent of skipping these top level fields was to reassemble them from noise handshake state since they are required at that level rather than duplicating them in the packet.

Reply:
Took a stab at this, lmk if it's what you had in mind.

Reply:
lgtm